_transcripts/011-jake-yoshua-stjepan.md

```diff
@@ -0,0 +1,1106 @@
+orderings. But also, you can put, like, vectors in `AtomicCell`s, and arbitrary
+kind of data.
+
+__Ben__: Is that, kind of, I know there's a crate called `once_cell` that uses
+`std::sync::Once` to initialize things exactly once. Is that kind of a
+complement to that, or a replacement for that?
+
+__Stjepan__: It's different. So `Once` is for initializing data once, whereas
+`AtomicCell` is like a cell, like the `Cell` type from the standard library.
+Except it's thread-safe, and you can update the contents of that cell as many
+times as you want. So not just once.
+
+__Ben__: Okay. When you took over the project, you said there were some things
+that were unsound about some of the APIs?
+
+__Stjepan__: Yes. There were some soundness issues in the epoch-based garbage
+collector, but I really won't go into that because it's super subtle.
+
+__Ben__: Do you feel like you've fixed those, since?
+
+__Stjepan__: Yes, those have been fixed.
+
+__Ben__: Okay. How do you even prove that kind of thing? I'm not going to ask
+you for, like, an actual proof, but like, for your own satisfaction, when did
+you feel like, okay, this is right now?
+
+__Stjepan__: I don't know. It's a lot of work and looking into the APIs and
+trying to find failure points. We also have this contributor, whose name is
+Jeehoon Kang, and he is a PhD researcher in Korea, specifically studying memory
+models and lock-free programming. And he made a proof of the correctness of our
+garbage collector.
+
+__Ben__: Very cool.
+
+__Stjepan__: Yeah.
+
+__Ben__: Excellent. That's great to have. I know back in the day, it was a big
+question like, well, we'd like to maybe include crossbeam in `std` one day, but
+we're not actually sure if it makes any sense, so it's fantastic to have that. I
+wanted to ask about it's use in `rayon`. So `rayon` uses the queue, the
+underlying queues, for its own purposes?
+
+__Stjepan__: `rayon` uses deques, work-stealing deques for balancing the
+workload among its own worker threads.
+
+__Ben__: Cool. So anyone using `rayon` is currently using `crossbeam`.
+
+__Stjepan__: Yes.
+
+__Ben__: Because `rayon` is just fantastic. Exemplary library. And then you also
+mentioned `async-std` is using `crossbeam`— I saw today, up in the presentation
+that Florian gave, about `async-std` being faster than `crossbeam` even,
+somehow? And then Yoshua told me to ask you about performance, in some way, and
+said that you working on performance, very heavily. Do you want to talk about
+that?
+
+__Stjepan__: So in `crossbeam`, we have our own channels which are based on
+`crossbeam` channels, in particular the bounded version of channels in
+`crossbeam`. So the performance advantage of `crossbeam` channels are in that,
+context switches on asynchronous tasks are much lower overhead than context
+switches with threads. So if you're sending a lot of data between different
+tasks, we can context switch between them much faster than with threads.
```

Comment:
I feel like there's something wrong with this part.  I don't know if it's me misunderstanding, or if the speaker just mis-spoke. At present the transcript effectively says "crossbeam channels are based on crossbeam channels", which doesn't make a lot of sense to me.

This paragraph starts at 50:29, if anyone wants to give it a listen and offer corrections.